www/js/alert-dialog.js

```diff
@@ -0,0 +1,97 @@
+
+
+define([
+	"dojo/_base/array",
+	"dojo/_base/declare",
+	"dojo/_base/lang",
+	"dojo/on",
+	"dojo/dom",
+	"dojo/dom-construct",
+	"dojo/query",
+	"dijit/registry",
+	"dojo/html",
+	"dojo/aspect",
+	"dijit/form/Button",
+], function(array, declare, lang, on, dom, domConstruct,  query, registry, html, aspect, Button) {
+	/* Summary:
+	 *				
+	 * 
+	 *			
+	 */
+	return declare(null, {
+		_clickHandlers: [],
+		_buttonsArray: [],
+
+		showDialog: function(title, alertContent, buttonsArray, cancelTitle){
+			alertDialog.set('title', title);
+			this._buttonsArray = buttonsArray;
+			d = dojo.byId('dialogButtons');
+			var cancelLabel = "Cancel";
+			if(cancelTitle && cancelTitle != ""){
+				cancelLabel = cancelTitle;
+			}
+
+			var closeButton = new Button({
+				label: cancelLabel,
+				id: 'cb'
+			});
+			var handler = on(closeButton, "click", lang.hitch(this, function(buttonsArray){
+				this.closeDialog(this._buttonsArray);
+			}));
+			
+			this._clickHandlers[closeButton]= handler;
+			domConstruct.place(closeButton.domNode, d, "last");
+			
+			var cb = dojo.byId('cb');
+
+			array.forEach(this._buttonsArray.reverse(), lang.hitch(this, function(button){
+				var b = new Button({
+					label: Object.keys(button)
+				});
+				
+				aspect.after(button,b.label[0],lang.hitch(this, function(response){
+ 					this.close();
+ 				}));
+				handler = on(b, "click", Object.values(button)[0]);
+				this._clickHandlers[button] = handler;
+				domConstruct.place(b.domNode, d, "first");
+				
+			}));
+
+			var dialogContent = "";
+			array.forEach(alertContent, function(message){
+				dialogContent += "<li>" + message + "</li>"
+			})	
+			
+			html.set(dojo.query('#alertDialog .dijitDialogPaneContentArea #content')[0], dialogContent);
+			alertDialog.show();
+		},
+
+		closeDialog: function(){
+			registry.byId("alertDialog").hide();
+		},
+		close: function(){
+			console.log("Close Called!! ");
+			if(window.history.length == 1)
+				window.close();
+			else
+				window.history.back();
```

Comment:
There was a requirement "Take as another argument an array of button text and function pairs where a button will be created for each. In addition to calling the function, all buttons should hide the dialog when clicked (so the functions don't have to hide it themselves)."
Mistook the requirement to have close called on every button click after performing its function. Made the necessary changes to call just close dialog after click executes and moved exit topomath code to ExitTopomath stub in main.js